src/components/common/CustomTable.js

```diff
@@ -146,23 +138,41 @@ const MembersTable = ({ members: rows, tableTitle, id: tableId, empty }) => {
   );
 
   // transform filteredRows' information into displayable information
-  const mappedRows = rowsToDisplay.map((item) => {
-    const { id, updatedAt, name, createdAt, type, email } = item;
-    const userNameAndIcon = (
-      <span className={classes.iconAndName}>
-        <AccountCircleIcon />
-        <span className={classes.itemName}>{name}</span>
-      </span>
-    );
+  const mappedRows = rowsToDisplay.map((row) => {
+    const allowed = headCells.map((cell) => cell.id);
+    allowed.push('id');
+    const display = _.pick(row, allowed);
+
+    if (iconCell) {
+      if (iconInfo) {
+        const iconComponentProps = _.pick(row, iconInfo);
+        display[iconCell] = (
+          <span className={classes.iconAndName}>
+            {React.cloneElement(icon, iconComponentProps)}
+            <span className={classes.itemName}>{display[iconCell]}</span>
+          </span>
+        );
+      } else {
+        display[iconCell] = (
+          <span className={classes.iconAndName}>
+            {icon}
+            <span className={classes.itemName}>{display[iconCell]}</span>
+          </span>
+        );
+      }
```

Comment:
I thin you can refactor by defining `icon` given `iconInfo`, and then set the component in a single html 
```
const icon = iconInfo ? ... : ...
display[iconCell] = ...
```